ui/i18n.py

```
@@ -0,0 +1,24 @@
+import gettext
+
+
+class I18n:
+    def __init__(self, locale='en_GB:en'):
+        if locale:
+            self.lang = locale
+        else:
+            self.lang = None
+        self.translate = None
+
+    def _(self, key):
+        if self.lang is not None:
+            try:
+                language = self.lang
+                translate = gettext.translation(
+                    'canute', localedir='ui/locale', languages=[language], fallback=False)
+                translate.install()
+            except Exception:
+                translate = gettext.NullTranslations()
```

You are swallowing a lot of potential errors here. It should at least be:
```python
except Exception as e:
    log.warning(e)
```

---

if not even `log.error(e)`